internal/test/e2e/migrations_test.go

```diff
@@ -109,8 +109,70 @@ func TestMigrations(t *testing.T) {
 			shardHosts += len(shard.Contracts)
 		}
 	}
+
 	// all shards should have 1 host except for 1. So we end up with 4 in total.
 	if shardHosts != 4 {
 		t.Fatalf("expected 4 shard hosts, got %v", shardHosts)
 	}
+
+	// create another bucket and add an object
+	tt.OK(b.CreateBucket(context.Background(), "newbucket", api.CreateBucketOptions{}))
+	tt.OKAll(w.UploadObject(context.Background(), bytes.NewReader(data), "newbucket", t.Name(), api.UploadObjectOptions{}))
+
+	// assert we currently don't have any alerts
+	ress, _ := b.Alerts(context.Background(), alerts.AlertsOpts{})
+	if ress.Totals.Error+ress.Totals.Critical != 0 {
+		t.Fatal("unexpected", ress)
+	}
+
+	// prepare
+
+	// remove all hosts to ensure migrations fail
+	for _, h := range cluster.hosts {
+		cluster.RemoveHost(h)
+	}
+
+	// fetch alerts and collect object ids until we found two
+	seen := make(map[types.Hash256]struct{})
+	got := make(map[string][]string)
+	tt.Retry(100, 100*time.Millisecond, func() error {
+		ress, _ := b.Alerts(context.Background(), alerts.AlertsOpts{})
+		if ress.Totals.Error+ress.Totals.Critical == 0 {
+			return errors.New("no migration alerts")
+		}
+		for _, alert := range ress.Alerts {
+			// skip if already seen
+			if _, skip := seen[alert.ID]; skip {
+				continue
+			}
+			seen[alert.ID] = struct{}{}
+
+			// skip if not a migration alert
+			data, ok := alert.Data["objectIDs"].(map[string]interface{})
+			if !ok {
+				continue
+			}
+
+			// collect all object ids per bucket
+			for bucket, ids := range data {
+				if objectIDs, ok := ids.([]interface{}); ok {
+					for _, id := range objectIDs {
+						got[bucket] = append(got[bucket], id.(string))
+						if len(got) == 2 {
```

Comment:
I'd check for the length after the alert loop to make sure we are not accidentally aborting too early. There might be another object in which case the test should fail.